src/extra/components/detector_motors.py

```diff
@@ -196,7 +196,10 @@ def find_motors(dc, pattern, position_key, data_selectors=None, **coordinates):
         src = pattern.format(**args)
         label = ''.join(f"{n}{v}" for n, v in args.items())
 
-        if src in dc.control_sources and position_key in dc[src].keys():
+        if (
+            src in dc.control_sources and
+            position_key in dc[src].keys(inc_timestamps=False)
```

Comment:
```suggestion
            position_key in dc[src]
```

This should work, and can be faster than loading the list of keys.